### PR TITLE
Add GitHub action CI builds

### PR DESCRIPTION
**Pull request by JakeShirley**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [ ] This PR addresses one or more open issues that were assigned to me:
      - No issues for this
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
This PR adds two new GitHub Action jobs:
- One that builds with `xcodebuild`
- One that tests with `xcodebuild`

It also disabled the UI tests project from running as it sounds like that is a waste of time and a stubbed out empty project.

## Screenshots and Videos
n/a

## Additional Context
This PR hard codes a specific device to build and test on in the iOS simulator (iPhone 14) as well as an iOS version string.

These will change over time when GitHub updates their MacOS agents but I added a step which prints available combinations so updating will be easy.  These values should be pretty stable (months between changing, not weeks) but we could probably automate the selection somehow with some fancy grep.

### CODE REVIEW

**Review by EricBAndrews — APPROVED** [review]
Free!

File: .github/workflows/ci_build.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    pull_request:
+        branches:
+        - master
+        - dev
+
+jobs:
+  Build:
+    permissions: write-all
+    runs-on: macos-latest
```
Comment (JakeShirley) [inline .github/workflows/ci_build.yml:17]:
```suggestion
    runs-on: macos-13
```

File: .github/workflows/ci_build.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    permissions: write-all
+    runs-on: macos-13
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - uses: maxim-lobanov/setup-xcode@v1
+      name: Set Xcode Version
+      with:
+        xcode-version: "14.2.0"
```
Comment (JakeShirley) [inline .github/workflows/ci_build.yml:25]:
```suggestion
        xcode-version: "14.3.1"
```

File: .github/workflows/ci_build.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        path: build_results.xcresult
+        upload-bundles: never # Permission issues with uploading
+        show-passed-tests: false
+      if: success() || failure()
+      # ^ This is important because the action will be run
+      # even if the test fails in the previous step.
+
+  Test:
+    permissions: write-all
+    runs-on: macos-latest
```
Comment (JakeShirley) [inline .github/workflows/ci_build.yml:52]:
```suggestion
    runs-on: macos-13
```

File: .github/workflows/ci_build.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - uses: maxim-lobanov/setup-xcode@v1
+      name: Set Xcode Version
+      with:
+        xcode-version: "14.2.0"
```
Comment (JakeShirley) [inline .github/workflows/ci_build.yml:60]:
```suggestion
        xcode-version: "14.3.1"
```
